frontend/server/src/main/java/com/amazonaws/ml/mms/wlm/ModelManager.java

```diff
@@ -162,24 +163,33 @@ public void workerStatus(final ChannelHandlerContext ctx) {
                     int numScaled = 0;
                     for (Map.Entry<String, Model> m : models.entrySet()) {
                         numScaled += m.getValue().getMinWorkers();
-                        numWorking += wlm.getWorkers(m.getValue().getModelName()).size();
+                        numWorking += wlm.getNumRunningWorkers(m.getValue().getModelName());
                     }
 
                     if ((numWorking > 0) && (numWorking < numScaled)) {
-                        // TODO: Check if this status can be set to
-                        // HttpResponseStatus.PARTIAL_CONTENT
                         response = "Partial Healthy";
                     } else if ((numWorking == 0) && (numScaled > 0)) {
-                        // TODO: Check if this can be set to HttpResponseStatus.NOT_FOUND;
                         response = "Unhealthy";
                     }
-                    NettyUtils.sendJsonResponse(ctx, new StatusResponse(response));
+
+                    // TODO: Check if its OK to send other 2xx errors to ALB for "Partial Healthy"
+                    // and "Unhealthy"
+                    NettyUtils.sendJsonResponse(
+                            ctx, new StatusResponse(response), HttpResponseStatus.OK);
                 };
         wlm.scheduleAsync(r);
     }
 
     public boolean scaleRequestStatus(String modelName) {
-        return wlm.getWorkerStatus(modelName);
+        Model model = ModelManager.getInstance().getModels().get(modelName);
+        int numWorkers = wlm.getNumRunningWorkers(modelName);
+        boolean success = true;
+
+        if ((model != null) && (model.getMinWorkers() > numWorkers)) {
```

Comment:
this can be simplified as:
return model == null || model.getMinWorkers() <= numWorkers;